File: test_debug.rs
use std::process::Command;

fn main() {
    let test_input = r#"fn a(first : I32) : (I32) => I32 => fn second(second : I32) => first + second; a(3)"#;
    
    // Try to interpret this manually
    println!("Testing: {}", test_input);
    
    // Compile and run the interpreter
    let output = Command::new("cargo")
        .args(&["run", "--", test_input])
        .current_dir("C:\\Users\\mathm\\IdeaProjects\\Tuff")
        .output();
    
    match output {
        Ok(output) => {
            println!("stdout: {}", String::from_utf8_lossy(&output.stdout));
            println!("stderr: {}", String::from_utf8_lossy(&output.stderr));
        }
        Err(e) => println!("Error: {}", e),
    }
}

File: test_debug.c
#include <stdio.h>
#include "interpret.h"

int main()
{
    printf("Testing: let mut x = 0; { if (true) x = 1; else x = 2; } x\n");
    InterpretResult result = interpret("let mut x = 0; { if (true) x = 1; else x = 2; } x");
    if (result.has_error)
    {
        printf("ERROR: %s\n", result.error_message);
    }
    else
    {
        printf("SUCCESS: Result = %d\n", result.value);
    }
    return 0;
}
